mailet/standard/src/test/java/org/apache/james/transport/mailets/ToProcessorTest.java

```diff
@@ -20,92 +20,118 @@
 
 package org.apache.james.transport.mailets;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
 import java.util.Arrays;
 
 import javax.mail.MessagingException;
-import javax.mail.internet.MimeMessage;
-import javax.mail.internet.ParseException;
 
 import org.apache.mailet.Mail;
 import org.apache.mailet.MailAddress;
 import org.apache.mailet.Mailet;
+import org.apache.mailet.MailetException;
 import org.apache.mailet.base.test.FakeMail;
 import org.apache.mailet.base.test.FakeMailContext;
 import org.apache.mailet.base.test.FakeMailetConfig;
-import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.slf4j.Logger;
 
 public class ToProcessorTest {
 
-    private MimeMessage mockedMimeMessage;
-
-    private Mail mockedMail;
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
 
     private Mailet mailet;
+    private FakeMailetConfig mailetConfig;
+    private Logger logger;
 
-    private String processor = null;
-
-    private String notice = null;
-
-    private void setProcessor(String processor) {
-        this.processor = processor;
+    @Before
+    public void setup() {
+        mailet = new ToProcessor();
+        logger = mock(Logger.class);
+        mailetConfig = new FakeMailetConfig("Test", FakeMailContext.builder().logger(logger).build());
     }
 
-    private void setNotice(String notice) {
-        this.notice = notice;
+    @Test
+    public void getMailetInfoShouldReturnValue() {
+        assertThat(mailet.getMailetInfo()).isEqualTo("ToProcessor Mailet");
     }
 
-    private void setupMockedMail(MimeMessage m) throws ParseException {
-        mockedMail = new FakeMail();
-        mockedMail.setMessage(m);
-        mockedMail.setRecipients(Arrays.asList(new MailAddress("test@james.apache.org"),
-                new MailAddress("test2@james.apache.org")));
+    @Test
+    public void initShouldThrowWhenProcessorIsNotGiven() throws MessagingException {
+        mailetConfig.setProperty("notice", "error in message");
+        expectedException.expect(MailetException.class);
 
+        mailet.init(mailetConfig);
     }
 
-    private void setupMailet() throws MessagingException {
-        mailet = new ToProcessor();
-        FakeMailetConfig mci = new FakeMailetConfig("Test",
-                FakeMailContext.defaultContext());
-        if (processor != null) {
-            mci.setProperty("processor", processor);
-        }
-        if (notice != null) {
-            mci.setProperty("notice", notice);
-        }
-        mailet.init(mci);
+    @Test
+    public void serviceShouldSetTheStateOfTheMail() throws MessagingException {
+        String processor = "error";
+        mailetConfig.setProperty("processor", processor);
+        mailet.init(mailetConfig);
+
+        Mail mail = new FakeMail();
```

Comment:
Ok, i will rebase my PR.
The bad point is that there's still a public constructor on FakeMail; and it's also not immutable.